### PR TITLE
Mgmt 4967 Create a standard way to run a debugger on the service

### PR DESCRIPTION
**Pull request by michaellevy101**
[MGMT-4967](https://issues.redhat.com/browse/MGMT-4967) Create a standard way to run a debugger on the service
Document how to debug deploy the service with debug information 
Open the debug ports on our k8s resources
Change the debug port to be configureable
 
 Fixing skipper make deploy-test command on a macos

### CODE REVIEW

**Review by pawanpinjarkar — COMMENTED** [review]
Looks good. Just one suggestion - please squash the 2 commits into one.

**Review by jakub-dzon — APPROVED** [review]
LGTM

**Comment by omertuc** [comment]
> Looks good. Just one suggestion - please squash the 2 commits into one.

Separation into multiple commits is something we do often to ease the burden on the reviewer. They automatically get squashed into a single commit (preserving the messages of both) during the automatic bot merge.

File: debug.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+Compile the code with debug information and patch the image:
+
+```shell
+skipper make update-debug-minimal
+```
+
+Compile the code with debug information, patch the image and push to your local k8s:
+
+```shell
+skipper make _update-local-image
```
Comment (omertuc) [inline debug.md:26]: Doesn't look like _update-local-image does any of `Compile the code with debug information, patch the image` ?

File: debug.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+export DEBUG_PORT=8765
+```
+
+Compile the code with debug information and patch the image:
+
+```shell
+skipper make update-debug-minimal
+```
+
+Compile the code with debug information, patch the image and push to your local k8s:
```
Comment (omertuc) [inline debug.md:23]: Maybe add a note about how the original image has to be built first, and then it gets patched with just debug binaries?

File: tools/deploy_inventory_service.py [hunk 3]
```diff
@@ -15,10 +17,20 @@ def main():
...
     with open(src_file, "r") as src:
         with open(dst_file, "w+") as dst:
-            data = src.read()
-            data = data.replace('REPLACE_NAMESPACE', f'"{deploy_options.namespace}"')
+            raw_data = src.read()
+            raw_data = raw_data.replace('REPLACE_NAMESPACE', f'"{deploy_options.namespace}"')
+            data = yaml.safe_load(raw_data)
+
+            if deploy_options.port:
+                for port_option in deploy_options.port:
```
Comment (omertuc) [inline tools/deploy_inventory_service.py:25]: nit: `for port_number, port_name in deploy_options.port:`

File: debug.md (hunk 1 above)
Comment (michaellevy101) [reply debug.md:26]: if you set the DEBUG env variable to true,  it's

File: debug.md (hunk 2 above)
Comment (michaellevy101) [reply debug.md:23]: Actually you don't have to build the original image you can just download it
The only problematic case is when you changed the image by setting the SERVICE environment variable and you never pushed it or the image in the remote registry is too old.
I think that when the image is not available we can easily just build it, but when the image itself change i'm not sure how to verify that

**Comment by omertuc** [comment]
> /hold

Why?

**Pull request merged**